engine/table/src/main/java/io/deephaven/engine/table/impl/DeferredViewTable.java

```diff
@@ -146,55 +146,86 @@ private PreAndPostFilters applyFilterRenamings(WhereFilter[] filters) {
         ArrayList<WhereFilter> preViewFilters = new ArrayList<>();
         ArrayList<WhereFilter> postViewFilters = new ArrayList<>();
 
+        final Set<String> postViewColumns = new HashSet<>();
+        final Map<String, String> renames = new HashMap<>();
+
+        // keep track of renames already applied, so that we can detect the case when a rename is renamed
+        final Map<String, Set<String>> reverseLookup = new HashMap<>();
+
+        // determine if filter has any dependencies on deferred columns
+        for (int ii = deferredViewColumns.length - 1; ii >= 0; --ii) {
+            // traverse columns in reverse order to handle scenarios where a deferred column is renamed
+            SelectColumn selectColumn = deferredViewColumns[ii];
+            final String columnName = selectColumn.getName();
+            if (selectColumn.isRetain()) {
+                continue;
+            }
+            if (selectColumn instanceof SwitchColumn) {
+                selectColumn = ((SwitchColumn) selectColumn).getRealColumn();
+            }
+
+            if (selectColumn instanceof SourceColumn) {
+                // this is a renamed column
+                final String innerName = ((SourceColumn) selectColumn).getSourceName();
+                final Set<String> myOuterNames = reverseLookup.computeIfAbsent(innerName, name -> new HashSet<>());
+
+                myOuterNames.add(columnName);
+                renames.put(columnName, innerName);
```

Comment:
I worry that we have a bug here. Since we're going backwards, are we possibly getting the wrong "final" rename if an outer name is used multiple times?

Reply:
Here's a unit test that fails in the way I was concerned about:
```
    @Test
    public void testMultipleRenamesWithSameOuterName() {
        testWriteRead(emptyTable(10).update("X = ii", "Y = ii", "Z = ii % 3"),
                t -> t.updateView("Y = Z", "Y = X").where("Y % 2 == 0"));
    }
```